### PR TITLE
Add e2e test for readOnlyRootFilesystem containers

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

This PR adds node e2e test for readOnlyRootFilesystem containers.

**Which issue this PR fixes**

Part of #44118.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e_node/security_context_test.go [hunk 1]
```diff
@@ -313,6 +313,66 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
+						},
+					},
+				},
+			}
+		}
+		createAndWaitUserPod := func(readOnlyRootFilesystem bool) string {
+			podName := fmt.Sprintf("busybox-readonly-%v-%s", readOnlyRootFilesystem, uuid.NewUUID())
+			podClient.Create(makeUserPod(podName,
+				"gcr.io/google_containers/busybox:1.24",
+				[]string{"sh", "-c", "touch checkfile && [ -f checkfile ] && echo Found || true"},
```
Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: Alternatively, you can simply run `"sh", "-c", "touch checkfile` and see whether the pod succeeded or failed. The upside is that we wouldn't need "logs" to check the state of the pod.

Comment (feiskyer) [inline test/e2e_node/security_context_test.go]: @yujuhong Yep, that's another way. I think both checking failure and logs are ok. And here using logs could be consistent with other places. I can change to checking failure if you prefer that way.

Comment (yujuhong) [inline test/e2e_node/security_context_test.go]: I think that's fine for now. We can try to minimize the reliance on logs in a separate PR if necessary.

**Comment by yujuhong** [comment]
/lgtm 

Leave this to @derekwaynecarr for approval.

**Pull request merged**